lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp

```diff
@@ -105,23 +105,35 @@ void MmeNasStateConverter::proto_to_hashtable_ts(
   }
 }
 
+char* MmeNasStateConverter::mme_app_convert_guti_to_string(guti_t* guti_p) {
+#define GUTI_STRING_LEN 20
+  char* str = (char*) calloc(1, sizeof(char) * GUTI_STRING_LEN);
+  snprintf(
+      str, GUTI_STRING_LEN, "%x%x%x%x%x%x%04x%x%08x",
```

Comment:
MME code is 8 bits (value range 0-255), which would use 2 hex chars, so it should be read as %02x. Note that we are okay reading each of mcc digits as 1 hex char as they are only using 4 bits.

Reply:
Oh! my bad, somehow did wrong calculation.
sure I will correct it